apps/web/src/components/common/form-input.tsx

```diff
@@ -10,24 +11,29 @@ import {
 import { Input } from '@packages/ui/components/input';
 import { LucideIcon } from 'lucide-react';
 import { InputHTMLAttributes } from 'react';
-import { FieldValues, Path, UseFormReturn } from 'react-hook-form';
+import { FieldValues, Path } from 'react-hook-form';
+
 
 type FormInputProps<T extends FieldValues> = {
   icon: LucideIcon;
-  form: UseFormReturn<T>;
   name: Path<T>;
   label?: string;
   description?: string;
+  validMessage?: string;
 } & Omit<InputHTMLAttributes<HTMLInputElement>, 'form' | 'name'>;
 
 export function FormInput<T extends FieldValues>({
   icon,
-  form,
   name,
   label,
+  validMessage,
   className,
   ...props
 }: FormInputProps<T>) {
+  const form = useFormContext<T>();
+  const { dirtyFields, errors } = form.formState;
+  const isFieldValid = dirtyFields[name as keyof typeof dirtyFields] && !errors[name];
```

Comment:
- 잘 동작하나요? 상위에 `FormProvider` 를 선언 안 하신것 같아 여쭤봅니다.
```suggestion
  const {formState} = useFormContext<T>();
  const { dirtyFields, errors } = formState;
  const isFieldValid = !errors[name];
```
- 이렇게만 써도 충분해요. `dirtyFields` 의 경우에는 required option 으로 error 에서 체크합니다.